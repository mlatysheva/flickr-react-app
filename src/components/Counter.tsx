import React, { useContext, useState } from 'react';
import { ThemeContext } from './Theme';
import { useAppSelector, useAppDispatch } from '../hooks';
import {
  decrement,
  increment,
  reset,
  incrementByAmount,
  incrementAsync,
} from '../features/counterSlice';

export const Counter = () => {
  const theme = useContext(ThemeContext);
  const count = useAppSelector((state) => state.counter.value);
  const dispatch = useAppDispatch();
  const [incrementAmount, setIncrementAmount] = useState('0');

  return (
    <div>
      <h1>Counter: {count}</h1>
      <div className="cards-wrapper">
        <button
          style={{ background: theme.state.background, color: theme.state.foreground }}
          onClick={(e: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => {
            console.log(`Clicked at: ${e.clientX}, ${e.clientY}`);
            dispatch(decrement());
          }}
        >
          -1
        </button>
        <button
          style={{ background: theme.state.background, color: theme.state.foreground }}
          onClick={() => dispatch(reset())}
        >
          Reset
        </button>
        <button
          style={{ background: theme.state.background, color: theme.state.foreground }}
          onClick={(e: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => {
            console.log(`Clicked at: ${e.clientX}, ${e.clientY}`);
            dispatch(increment());
          }}
        >
          +1
        </button>
        <input
          className="input-amount"
          aria-label="Set increment amount"
          value={incrementAmount}
          onChange={(e) => setIncrementAmount(e.target.value)}
        />
        <button onClick={() => dispatch(incrementByAmount(Number(incrementAmount) || 0))}>
          Add Amount
        </button>
        <button onClick={() => dispatch(incrementAsync(Number(incrementAmount) || 0))}>
          Add Async
        </button>
      </div>
    </div>
  );
};
