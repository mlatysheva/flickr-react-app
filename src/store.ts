import { configureStore } from '@reduxjs/toolkit';
import apiSearchReducer from './features/apiSearchSlice';
import counterReducer from './features/counterSlice';
import photosReducer from './features/photoSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    apiSearch: apiSearchReducer,
    photos: photosReducer,
  },
});

export const counter = (state: { counter: { value: number } }) => state.counter.value;

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;
