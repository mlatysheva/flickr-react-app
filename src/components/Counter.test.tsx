import { render, screen } from '@testing-library/react';
import { Counter } from './Counter';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../store';

describe("Counter", () => {
  it("renders Counter component", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Counter />
        </MemoryRouter>
      </Provider>);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText("+1")).toBeInTheDocument();
    expect(screen.getByRole('button')).toBeInTheDocument();
  })
})