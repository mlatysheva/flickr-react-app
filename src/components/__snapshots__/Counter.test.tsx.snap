// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders Counter component 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Counter: 0
    </h1>
    <button>
      +1
    </button>
  </div>
</DocumentFragment>
`;
